File: 백준/Silver/2606. 바이러스/바이러스.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;
import java.util.StringTokenizer;

public class Main {
	static List<Integer>[] graph;
	static boolean[] visited;
	static int count = 0;

	public static void main(String[] args) throws IOException {

		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		int n = Integer.parseInt(br.readLine());
		int m = Integer.parseInt(br.readLine());

		graph = new ArrayList[n + 1];
		for (int i = 0; i <= n; i++) {
			graph[i] = new ArrayList<>();
		}

		visited = new boolean[n + 1];

		for (int i = 0; i < m; i++) {
			StringTokenizer st = new StringTokenizer(br.readLine());
			int u = Integer.parseInt(st.nextToken());
			int v = Integer.parseInt(st.nextToken());

			graph[u].add(v);
			graph[v].add(u);
		}

		dfs(1);

		System.out.println(count - 1);
	}

	static void dfs(int node) {
		visited[node] = true;
		count++;

		for (int next : graph[node]) {
			if (!visited[next]) {
				dfs(next);
			}
		}
	}
}

File: 백준/Silver/2606. 바이러스/바이러스.py
import sys

sys.setrecursionlimit(10**6)

n = int(sys.stdin.readline().rstrip())
m = int(sys.stdin.readline().rstrip())
graph = [[] for _ in range(n + 1)]
visited = [False] * (n + 1)
cnt = 0


def dfs(node):
    global cnt
    visited[node] = True
    cnt += 1

    for next in graph[node]:
        if not visited[next]:
            dfs(next)


for i in range(m):
    u, v = map(int, sys.stdin.readline().rstrip().split(" "))
    graph[u].append(v)
    graph[v].append(u)

dfs(1)

print(cnt - 1)
